.github/workflows/ci.yml

```diff
@@ -0,0 +1,159 @@
+name: ci
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  RUST_PREVIOUS_VERSION: 1.86.0
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  # Enable consistent use of `RUST_PREVIOUS_VERSION` across different jobs
+  prepare-checks:
+    runs-on: ubuntu-latest
+
+    outputs:
+      rust-prev-version: ${{ env.RUST_PREVIOUS_VERSION }}
+
+    steps:
+      - name: Set rust-prev-version output
+        run: echo "rust-prev-version=${RUST_PREVIOUS_VERSION}" >> $GITHUB_OUTPUT
+
+################################## FORMAT AND LINTS LAYER ######################
+
+  clippy-rustfmt:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+          components: clippy, rustfmt
+
+      - name: Run cargo clippy
+        run: cargo clippy --all-targets -- -D warnings
```

Comment:
Let's use `pedantic` options: ` cargo clippy --all-targets -- -Dclippy::all -Dclippy::pedantic` in this way we also analyse `tests` and `examples` for each crate of the workspace. Feel free to fix all internal lints as you wish, but do not touch the lints related to public interfaces. Let's use `[lint]` option to allow them or the `#[allow(...)]` attribute